Sources/Datadog/Core/System/Time/ServerDateCorrector.swift

```diff
@@ -10,8 +10,10 @@ import Foundation
 internal class ServerDateCorrector: DateCorrector {
     /// Server offset publisher.
     private let publisher: ValuePublisher<TimeInterval> = ValuePublisher(initialValue: 0)
+    private let provider: ServerDateProvider
 
     init(serverDateProvider: ServerDateProvider) {
+        self.provider = serverDateProvider
```

Comment:
Now that Kronos no longer retains `self` in completion, we need to keep a strong ref to the provider.